nilearn/plotting/surf_plotting.py

```diff
@@ -369,30 +381,47 @@ def _get_cmap_matplotlib(cmap, vmin, vmax, threshold=None):
 
 
 def _compute_facecolors_matplotlib(bg_map, faces, n_vertices,
-                                   darkness, alpha):
+                                   darkness, alpha, bg_map_rescale):
     """Helper function for plot_surf with matplotlib engine.
 
     This function computes the facecolors.
     """
     # set alpha if in auto mode
     if alpha == 'auto':
         alpha = .5 if bg_map is None else 1
+
     if bg_map is None:
         bg_data = np.ones(n_vertices) * 0.5
     else:
-        bg_data = load_surf_data(bg_map)
+        # bg_data = np.copy(load_surf_data(bg_map))
+        bg_data = np.copy(load_surf_data(bg_map))
         if bg_data.shape[0] != n_vertices:
             raise ValueError('The bg_map does not have the same number '
                              'of vertices as the mesh.')
+
     bg_faces = np.mean(bg_data[faces], axis=1)
-    if bg_faces.min() != bg_faces.max():
-        bg_faces = bg_faces - bg_faces.min()
-        bg_faces = bg_faces / bg_faces.max()
-    # control background darkness
-    bg_faces *= darkness
+    # scale background map if need be
+    bg_vmin, bg_vmax = np.min(bg_faces), np.max(bg_faces)
+    if (
+        bg_map_rescale is True
+        or (
+            isinstance(bg_map_rescale, str)
+            and bg_map_rescale == "auto"
+            and (bg_vmin < 0 or bg_vmax > 1)
+        )
+    ):
+        bg_norm = mpl.colors.Normalize(vmin=bg_vmin, vmax=bg_vmax)
+        bg_faces = bg_norm(bg_faces)
+
+    if darkness is not None:
+        bg_data *= darkness
```

Comment:
here this should be `bg_faces`, not `bg_data` -- as a result the `darkness` parameter is broken for the matplotlib engine

Reply:
Good catch, sorry about that!